pkg/spdk/server.go

```diff
@@ -206,6 +215,130 @@ func (s *Server) EngineSnapshotDelete(ctx context.Context, req *spdkrpc.Snapshot
 	return &empty.Empty{}, err
 }
 
+func (s *Server) DiskCreate(ctx context.Context, req *spdkrpc.DiskCreateRequest) (*spdkrpc.Disk, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"diskName":  req.DiskName,
+		"diskPath":  req.DiskPath,
+		"blockSize": req.BlockSize,
+	})
+
+	log.Info("Creating disk")
+
+	if req.DiskName == "" || req.DiskPath == "" {
+		return nil, grpcstatus.Error(grpccodes.InvalidArgument, "disk name and disk path are required")
+	}
+
+	s.Lock()
+	defer s.Unlock()
```

Comment:
Do we need this lock? I am thinking if this lock is necessary for using `s.spdkClient`

Reply:
Not really sure if the spdk_tgt respond in order...